src/subject_name/mod.rs

```diff
@@ -13,6 +13,8 @@
 // OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
 mod dns_name;
+#[cfg(feature = "alloc")]
+pub(crate) use dns_name::GeneralDnsNameRef;
```

Comment:
This really needs exposed here and in lib.rs or otherwise I can't do something like this:

```rust
let dns_names: Vec<String> = cert
    .dns_names()?
    .filter_map(|name| match name {
        GeneralDnsNameRef::DnsName(name) => {
            let name: &str = name.into();
            Some(name.to_string())
        }
        GeneralDnsNameRef::Wildcard(_) => None,
    })
    .collect();
```

Alternatively `GeneralDnsNameRef` could have a `fn is_wildcard(&self) -> bool` on it, but as-is I don't see a way to tell wildcard from not, which pretty much removes the reason to separate these with the type system all together?

Thanks for the work on this by the way!

Reply:
Thanks for taking a look!

> as-is I don't see a way to tell wildcard from not, which pretty much removes the reason to separate these with the type system all together?

I think the motivation for the separation is for differentiating between the two in the context of name verification, and the `dns_names` function explicitly mentions that it's **not** to be used to implement custom name verification.

Can you expand on your use-case for wanting to filter out wildcard names from `dns_names`? I see it as a diagnostic addition to the API for the case where you want to display the certificate's names to a user in an informational capacity. In that context I'm not sure there's a need to be able to separate out wildcards.

Reply:
My use-case is a TLS tunnel where I'm routing incoming connections to the proper backend via SNI, so if a backend cert has names `bob.com *.example.org`, and the incoming connection SNI is `test.example.org` I have to be able to know the difference because I have to lookup wildcards separately from full domains.

Does that make sense?  I will eventually push up this code but it might be a few more weeks...

Reply:
To get a little more concrete, I have 2 `Arc<RwLock<HashMap<String, Connection>>>`, which is a dns_name -> Connection mapping, one for FQDNs and one for Wildcards, so a TLS connection comes in, I read SNI, look up the right connection via one of these 2 maps, and start shuffling bytes.

Reply:
> My use-case is a TLS tunnel where I'm routing incoming connections to the proper backend via SNI, so if a backend cert has names bob.com *.example.org, and the incoming connection SNI is test.example.org I have to be able to know the difference because I have to lookup wildcards separately from full domains.

I think matching a reference DNS name from SNI to an end entity certificate is already handled by [EndEntityCert.verify_is_valid_for_dns_name](https://docs.rs/webpki/latest/webpki/struct.EndEntityCert.html#method.verify_is_valid_for_dns_name). It already takes into account wildcards (and the other peculiarities of doing this task safely) and wouldn't require you to make the distinction yourself.

Reply:
Only by looping through thousands of certificates and calling this method for each on every new incoming connection? That's why I need the quick lookup of a hashmap instead.

Reply:
Please open up a new issue where we can discuss your use case. We're certainly open to considering it but I don't think it makes sense to block this PR (which has different goals).

Reply:
Absolutely didn't want to block this PR, thanks, will do!